File: Java/GenericStructures/Vector.java
package GenericStructures;

public class Vector <T>{
    private T [] ptr_arr;
    private int _size;
    private int _counter;

    public Vector() {
        _size = 1;
        _counter = 0;
        ptr_arr = (T[]) new Object[this._size];
    }

    public void add(T data){
        _counter += 1;
        if(_counter == _size) this.grow(); 
        ptr_arr[_counter] = data;
    }

    private void grow(){
        T [] temp = null;
        int size = _size * 2;
        temp = (T[]) new Object[this._size];

        for(int i = 0; i < this._size; ++i)
            temp[i] = ptr_arr[i];
        
        ptr_arr = temp;
        _size = size;
    }

    public void display(){
        for(int i = 0; i < this._size; ++i)
            System.out.print(String.valueOf(ptr_arr[i]) + " ");
    }
}

class MainVector{
    public static void main(String [] args){

    }
}

File: C#/GenericStructures/Vector.cs
using System;

class Vector<T>{
    T [] ptr = null;
    int m_size = 1; 
    int counter = 0;

    public Vector(int size){
        this.ptr = new T[size];
    }

    public Vector(){
        this.ptr  = new T[1];
    }
    
    public void add(T data){
        if(this.counter == this.m_size) grow();
        this.ptr[counter] = data;
        ++ this.counter;
    }

    public void display(){
        foreach(int i in this.ptr[])
            Console.Write(this.ptr[i] + " ");
    }
    
    private void grow(){
        int newm_size = this.m_size * 2;
        T [] temp = new T[newm_size];

        for(int i = 0; i < this.m_size; ++i)
            temp[i] = this.ptr[i];
        
        this.ptr = temp;
        this.m_size = newm_size;
    }
}

class MainVector{
    public static void Main(string [] args){
        Vector a = new Vector();
        for(int i = 0; i<10; ++i)
            a.add(i);
        a.display();
    }
}